From 55fb15c78cbb5b0a43181b3d6014eef8da69fef8 Mon Sep 17 00:00:00 2001
From: "Tom G. Christensen" <tgc@jupiterrise.com>
Date: Fri, 3 Jun 2022 21:46:42 +0200
Subject: [PATCH 01/12] Ensure INET_ADDRSTRLEN is defined

---
 git-compat-util.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/git-compat-util.h b/git-compat-util.h
index bebcf9f698..debf09ae2d 100644
--- a/git-compat-util.h
+++ b/git-compat-util.h
@@ -1099,3 +1099,10 @@ extern int not_supposed_to_survive;
 #endif /* CHECK_ASSERTION_SIDE_EFFECTS */
 
 #endif
+
+#if !defined(INET_ADDRSTRLEN)
+#define INET_ADDRSTRLEN 16
+#endif
+#if !defined(INET6_ADDRSTRLEN) && !defined(NO_IPV6)
+#define INET6_ADDRSTRLEN 46
+#endif
-- 
2.43.7

