From cfefeb95a678ad370299e7127b87fa854e4ccfea Mon Sep 17 00:00:00 2001
From: "Tom G. Christensen" <tgc@jupiterrise.com>
Date: Sun, 5 Jun 2022 10:45:03 +0200
Subject: [PATCH 02/12] Ensure SCNuMAX is defined

---
 git-compat-util.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/git-compat-util.h b/git-compat-util.h
index debf09ae2d..c4a231e79d 100644
--- a/git-compat-util.h
+++ b/git-compat-util.h
@@ -1106,3 +1106,7 @@ extern int not_supposed_to_survive;
 #if !defined(INET6_ADDRSTRLEN) && !defined(NO_IPV6)
 #define INET6_ADDRSTRLEN 46
 #endif
+
+#ifndef SCNuMAX
+#define SCNuMAX PRIuMAX
+#endif
-- 
2.43.7

