From 535848f4ed2111fc21f43e18e305ca29415ecd0e Mon Sep 17 00:00:00 2001
From: "Tom G. Christensen" <tgc@jupiterrise.com>
Date: Sun, 5 Jun 2022 22:04:22 +0200
Subject: [PATCH 09/12] Use better shell in t5801 helper

---
 t/t5801/git-remote-testgit | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/t5801/git-remote-testgit b/t/t5801/git-remote-testgit
index f8b476499f..a7638be20c 100755
--- a/t/t5801/git-remote-testgit
+++ b/t/t5801/git-remote-testgit
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/ksh
 # Copyright (c) 2012 Felipe Contreras
 
 # The first argument can be a url when the fetch/push command was a url
-- 
2.43.7

