From 85cb245354e6a38ea30074fa8a86afaa22e164e1 Mon Sep 17 00:00:00 2001
From: "Tom G. Christensen" <tgc@jupiterrise.com>
Date: Sun, 5 Jun 2022 21:46:51 +0200
Subject: [PATCH 11/12] UTC is GMT on Solaris < 8

---
 t/t0006-date.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/t0006-date.sh b/t/t0006-date.sh
index 53ced36df4..1a29ff7e3a 100755
--- a/t/t0006-date.sh
+++ b/t/t0006-date.sh
@@ -63,7 +63,7 @@ check_show unix-local "$TIME" '1466000000'
 check_show 'format:%z' "$TIME" '+0200'
 check_show 'format-local:%z' "$TIME" '+0000'
 check_show 'format:%Z' "$TIME" ''
-check_show 'format-local:%Z' "$TIME" 'UTC'
+check_show 'format-local:%Z' "$TIME" 'GMT'
 check_show 'format:%%z' "$TIME" '%z'
 check_show 'format-local:%%z' "$TIME" '%z'
 
-- 
2.43.7

