--- git-1.5.5.1/git-compat-util.h.socklen_t	2008-05-11 20:17:59.225343000 +0200
+++ git-1.5.5.1/git-compat-util.h	2008-05-11 20:20:01.765932000 +0200
@@ -453,4 +453,8 @@
 # define FORCE_DIR_SET_GID 0
 #endif
 
+#ifdef NO_SOCKLEN_T
+typedef int socklen_t;
+#endif
+
 #endif
--- git-1.5.5.1.hack/Makefile.socklen_t	2008-05-11 20:26:52.657848000 +0200
+++ git-1.5.5.1.hack/Makefile	2008-05-11 20:28:06.418161000 +0200
@@ -852,6 +852,10 @@
 	LIB_OBJS += compat/inet_pton.o
 endif
 
+ifdef NO_SOCKLEN_T
+	BASIC_CFLAGS += -DNO_SOCKLEN_T
+endif
+
 ifdef NO_ICONV
 	BASIC_CFLAGS += -DNO_ICONV
 endif
