--- git-2.6.3/compat/regex/regcomp.c.no_stdint-h	2015-11-05 22:31:55.000000000 +0100
+++ git-2.6.3/compat/regex/regcomp.c	2015-11-11 18:24:26.206399000 +0100
@@ -18,7 +18,7 @@
    Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
    02110-1301 USA.  */
 
-#include <stdint.h>
+#include <inttypes.h>
 
 static reg_errcode_t re_compile_internal (regex_t *preg, const char * pattern,
 					  size_t length, reg_syntax_t syntax);
