--- git-1.5.5.1/Makefile.sunos56	2008-05-12 17:05:44.916828000 +0200
+++ git-1.5.5.1/Makefile	2008-05-12 17:06:54.787133000 +0200
@@ -573,6 +573,21 @@
 	NO_MEMMEM = YesPlease
 	NO_HSTRERROR = YesPlease
 	NO_MKDTEMP = YesPlease
+	ifeq ($(uname_R),5.6)
+		NEEDS_LIBICONV = YesPlease
+		NO_UNSETENV = YesPlease
+		NO_SETENV = YesPlease
+		NO_C99_FORMAT = YesPlease
+		NO_STRTOUMAX = YesPlease
+		NO_D_TYPE_IN_DIRENT = YesPlease
+		NO_SOCKADDR_STORAGE = YesPlease
+		NO_STRCASESTR = YesPlease
+		NO_STRLCPY = YesPlease
+		NO_STRTOUMAX = YesPlease
+		NO_IPV6 = YesPlease
+		NO_SOCKLEN_T = YesPlease
+		EXTLIBS += -lresolv
+	endif
 	ifeq ($(uname_R),5.8)
 		NEEDS_LIBICONV = YesPlease
 		NO_UNSETENV = YesPlease
