diff -up diffutils-3.3/po/Makefile.in.in.orig diffutils-3.3/po/Makefile.in.in
--- diffutils-3.3/po/Makefile.in.in.orig	2013-03-23 15:51:10.000000000 +0100
+++ diffutils-3.3/po/Makefile.in.in	2013-10-22 20:50:20.794686000 +0200
@@ -42,7 +42,7 @@ INSTALL_DATA = @INSTALL_DATA@
 mkinstalldirs = $(SHELL) @install_sh@ -d
 install_sh = $(SHELL) @install_sh@
 MKDIR_P = @MKDIR_P@
-mkdir_p = @mkdir_p@
+mkdir_p = $(MKDIR_P)
 
 GMSGFMT_ = @GMSGFMT@
 GMSGFMT_no = @GMSGFMT@
