--- expat-2.5.0/xmlwf/xmltchar.h.strtof	2022-10-24 18:32:57.000000000 +0000
+++ expat-2.5.0/xmlwf/xmltchar.h	2023-01-15 13:10:45.819376000 +0000
@@ -74,6 +74,6 @@
 #  define tmain main
 #  define tremove remove
 #  define tchar char
-#  define tcstof strtof
+#  define tcstof strtod
 #  define tcstoull strtoull
 #endif /* not XML_UNICODE */
--- expat-2.5.0/tests/runtests.c.strtof	2023-01-15 13:09:44.309378000 +0000
+++ expat-2.5.0/tests/runtests.c	2023-01-15 13:10:18.989378000 +0000
@@ -12002,12 +12002,12 @@
 
 static float
 portableNAN(void) {
-  return strtof("nan", NULL);
+  return strtod("nan", NULL);
 }
 
 static float
 portableINFINITY(void) {
-  return strtof("infinity", NULL);
+  return strtod("infinity", NULL);
 }
 
 START_TEST(test_billion_laughs_attack_protection_api) {
