diff -up expat-2.2.9/lib/siphash.h.orig expat-2.2.9/lib/siphash.h
--- expat-2.2.9/lib/siphash.h.orig	2019-09-21 17:23:24.000000000 +0000
+++ expat-2.2.9/lib/siphash.h	2020-03-24 17:10:54.980006000 +0000
@@ -103,7 +103,7 @@ typedef unsigned __int8 uint8_t;
 typedef unsigned __int32 uint32_t;
 typedef unsigned __int64 uint64_t;
 #else
-#  include <stdint.h> /* uint64_t uint32_t uint8_t */
+#  include <inttypes.h> /* uint64_t uint32_t uint8_t */
 #endif
 
 /*
