diff -up sudo-1.8.12/plugins/sudoers/Makefile.in.netlibs sudo-1.8.12/plugins/sudoers/Makefile.in
--- sudo-1.8.12/plugins/sudoers/Makefile.in.netlibs	2015-01-30 20:53:59.000000000 +0100
+++ sudo-1.8.12/plugins/sudoers/Makefile.in	2015-03-20 21:48:53.683094000 +0100
@@ -54,7 +54,7 @@ LT_LIBS = $(top_builddir)/lib/util/libsu
 LIBS = $(LT_LIBS) @LIBINTL@
 NET_LIBS = @NET_LIBS@
 SUDOERS_LIBS = @SUDOERS_LIBS@ @AFS_LIBS@ @GETGROUPS_LIB@ $(LIBS) $(NET_LIBS) @ZLIB@ @LIBMD@
-REPLAY_LIBS = @REPLAY_LIBS@ @ZLIB@
+REPLAY_LIBS = @REPLAY_LIBS@ $(NET_LIBS) @ZLIB@
 VISUDO_LIBS = $(NET_LIBS) @LIBMD@
 TESTSUDOERS_LIBS = $(NET_LIBS) @LIBMD@
 
diff -up sudo-1.8.12/lib/util/Makefile.in.netlibs sudo-1.8.12/lib/util/Makefile.in
--- sudo-1.8.12/lib/util/Makefile.in.netlibs	2015-03-21 10:45:11.659507000 +0100
+++ sudo-1.8.12/lib/util/Makefile.in	2015-03-21 10:46:59.430453000 +0100
@@ -80,7 +80,7 @@ CPPCHECK_OPTS = -q --force --enable=warn
 # Regression tests
 TEST_PROGS = atofoo_test conf_test hltq_test parseln_test progname_test @COMPAT_TEST_PROGS@
 TEST_LIBS = @LIBS@ @LIBINTL@
-TEST_LDFLAGS = @LDFLAGS@
+TEST_LDFLAGS = @LDFLAGS@ @NET_LIBS@
 
 # User and group ids the installed files should be "owned" by
 install_uid = 0
