diff -up sudo-1.8.6p5/plugins/sudoers/Makefile.in.ssp sudo-1.8.6p5/plugins/sudoers/Makefile.in
--- sudo-1.8.6p5/plugins/sudoers/Makefile.in.ssp	2013-01-11 20:38:37.000000000 +0100
+++ sudo-1.8.6p5/plugins/sudoers/Makefile.in	2013-05-09 12:16:36.331635000 +0200
@@ -187,7 +187,7 @@ libparsesudoers.la: $(LIBPARSESUDOERS_OB
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(LIBPARSESUDOERS_OBJS) -no-install
 
 sudoers.la: $(SUDOERS_OBJS) $(LT_LIBS) libparsesudoers.la @LT_LDDEP@
-	$(LIBTOOL) @LT_STATIC@ --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(SUDOERS_OBJS) libparsesudoers.la $(SUDOERS_LIBS) -module -avoid-version -rpath $(plugindir)
+	$(LIBTOOL) @LT_STATIC@ --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(SUDOERS_OBJS) $(SSP_LDFLAGS) libparsesudoers.la $(SUDOERS_LIBS) -module -avoid-version -rpath $(plugindir)
 
 visudo: libparsesudoers.la $(VISUDO_OBJS) $(LT_LIBS)
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(VISUDO_OBJS) $(LDFLAGS) $(PIE_LDFLAGS) $(SSP_LDFLAGS) libparsesudoers.la $(LIBS) $(NET_LIBS)
diff -up sudo-1.8.6p5/src/Makefile.in.ssp sudo-1.8.6p5/src/Makefile.in
--- sudo-1.8.6p5/src/Makefile.in.ssp	2013-01-11 20:36:54.000000000 +0100
+++ sudo-1.8.6p5/src/Makefile.in	2013-05-09 12:28:23.950272000 +0200
@@ -109,7 +109,7 @@ sudo: $(OBJS) $(LT_LIBS)
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJS) $(LDFLAGS) $(PIE_LDFLAGS) $(SSP_LDFLAGS) $(LIBS)
 
 libsudo_noexec.la: sudo_noexec.lo
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ sudo_noexec.lo -avoid-version -rpath $(noexecdir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ sudo_noexec.lo $(SSP_LDFLAGS) -avoid-version -rpath $(noexecdir)
 
 sesh: sesh.o error.o exec_common.o @LIBINTL@ $(LT_LIBS)
 	$(LIBTOOL) --mode=link $(CC) -o $@ sesh.o error.o exec_common.o $(LDFLAGS) $(PIE_LDFLAGS) $(SSP_LDFLAGS) @LIBINTL@ $(LIBS)
diff -up sudo-1.8.6p5/plugins/sample/Makefile.in.ssp sudo-1.8.6p5/plugins/sample/Makefile.in
--- sudo-1.8.6p5/plugins/sample/Makefile.in.ssp	2013-01-11 20:36:52.000000000 +0100
+++ sudo-1.8.6p5/plugins/sample/Makefile.in	2013-05-09 12:36:41.506755000 +0200
@@ -102,7 +102,7 @@ $(shlib_opt): $(shlib_exp)
 	@sed 's/^/+e /' $(shlib_exp) > $@
 
 sample_plugin.la: $(OBJS) @LT_LDDEP@
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version -rpath $(plugindir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(SSP_LDFLAGS) $(LIBS) -module -avoid-version -rpath $(plugindir)
 
 pre-install:
 
diff -up sudo-1.8.6p5/plugins/sample_group/Makefile.in.ssp sudo-1.8.6p5/plugins/sample_group/Makefile.in
--- sudo-1.8.6p5/plugins/sample_group/Makefile.in.ssp	2013-01-11 20:36:52.000000000 +0100
+++ sudo-1.8.6p5/plugins/sample_group/Makefile.in	2013-05-09 12:51:58.940640000 +0200
@@ -103,7 +103,7 @@ $(shlib_opt): $(shlib_exp)
 	@sed 's/^/+e /' $(shlib_exp) > $@
 
 sample_group.la: $(OBJS) $(LT_LIBS) @LT_LDDEP@
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version -rpath $(plugindir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(SSP_LDFLAGS) $(LIBS) -module -avoid-version -rpath $(plugindir)
 
 pre-install:
 
diff -up sudo-1.8.6p5/plugins/system_group/Makefile.in.ssp sudo-1.8.6p5/plugins/system_group/Makefile.in
--- sudo-1.8.6p5/plugins/system_group/Makefile.in.ssp	2013-01-11 20:36:53.000000000 +0100
+++ sudo-1.8.6p5/plugins/system_group/Makefile.in	2013-05-09 12:52:41.140375000 +0200
@@ -103,7 +103,7 @@ $(shlib_opt): $(shlib_exp)
 	@sed 's/^/+e /' $(shlib_exp) > $@
 
 system_group.la: $(OBJS) $(LT_LIBS) @LT_LDDEP@
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(LIBS) -module -avoid-version -rpath $(plugindir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) $(LT_LDFLAGS) -o $@ $(OBJS) $(SSP_LDFLAGS) $(LIBS) -module -avoid-version -rpath $(plugindir)
 
 pre-install:
 
