*** bzip2-1.0.5/Makefile-libbz2_so.orig	Sat Apr 26 11:42:11 2008
--- bzip2-1.0.5/Makefile-libbz2_so	Sat Apr 26 11:45:11 2008
***************
*** 35,41 ****
        bzlib.o
  
  all: $(OBJS)
! 	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.4 $(OBJS)
  	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.4
  	rm -f libbz2.so.1.0
  	ln -s libbz2.so.1.0.4 libbz2.so.1.0
--- 35,41 ----
        bzlib.o
  
  all: $(OBJS)
! 	$(CC) -shared -Wl,-h -Wl,libbz2.so.1 -o libbz2.so.1.0.4 $(OBJS)
  	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.4
  	rm -f libbz2.so.1.0
  	ln -s libbz2.so.1.0.4 libbz2.so.1.0
