diff -up gcc-3.4.6/gcc/configure.makeinfo gcc-3.4.6/gcc/configure
--- gcc-3.4.6/gcc/configure.makeinfo	2012-09-09 12:08:11.130034000 +0200
+++ gcc-3.4.6/gcc/configure	2012-09-09 12:17:43.791663000 +0200
@@ -5232,7 +5232,7 @@ else
   echo "configure:5232: version of makeinfo is $ac_prog_version" >&5
   case $ac_prog_version in
     '')     gcc_cv_prog_makeinfo_modern=no;;
-    4.[2-9]*)
+    4.[7-9]*|4.[1-9][0-9]*|[5-9]*|[1-9][0-9]*)
             gcc_cv_prog_makeinfo_modern=yes;;
     *)      gcc_cv_prog_makeinfo_modern=no;;
   esac
