--- grep-2.5/src/Makefile.in.orig	2004-01-26 14:31:58.850016000 +0100
+++ grep-2.5/src/Makefile.in	2004-01-26 14:32:48.840007000 +0100
@@ -354,8 +354,8 @@
 
 
 install-exec-local:
-	@SYMLINK@ @SYMLINKFROM@ $(bindir)/egrep
-	@SYMLINK@ @SYMLINKFROM@ $(bindir)/fgrep
+	@SYMLINK@ @SYMLINKFROM@ $(DESTDIR)$(bindir)/egrep
+	@SYMLINK@ @SYMLINKFROM@ $(DESTDIR)$(bindir)/fgrep
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
