diff -up openssl-1.1.1d/crypto/bio/b_addr.c.orig openssl-1.1.1d/crypto/bio/b_addr.c
--- openssl-1.1.1d/crypto/bio/b_addr.c.orig	2019-09-10 15:13:07.000000000 +0000
+++ openssl-1.1.1d/crypto/bio/b_addr.c	2020-02-07 21:09:45.902671700 +0000
@@ -847,7 +847,7 @@ int BIO_lookup_ex(const char *host, cons
             }
 
             if (endp != service && *endp == '\0'
-                    && portnum > 0 && portnum < 65536) {
+                    && portnum >= 0 && portnum < 65536) {
                 se_fallback.s_port = htons((unsigned short)portnum);
                 se_fallback.s_proto = proto;
                 se = &se_fallback;
