diff -up openssl-1.1.1d/apps/s_socket.c.inet6 openssl-1.1.1d/apps/s_socket.c
--- openssl-1.1.1d/apps/s_socket.c.inet6	2019-09-10 15:13:07.000000000 +0000
+++ openssl-1.1.1d/apps/s_socket.c	2020-01-28 20:54:45.186856000 +0000
@@ -160,7 +160,9 @@ int init_client(int *sock, const char *h
     if (*sock == INVALID_SOCKET) {
         if (bindaddr != NULL && !found) {
             BIO_printf(bio_err, "Can't bind %saddress for %s%s%s\n",
+#ifdef AF_INET6
                        BIO_ADDRINFO_family(res) == AF_INET6 ? "IPv6 " :
+#endif
                        BIO_ADDRINFO_family(res) == AF_INET ? "IPv4 " :
                        BIO_ADDRINFO_family(res) == AF_UNIX ? "unix " : "",
                        bindhost != NULL ? bindhost : "",
@@ -237,6 +239,7 @@ int do_server(int *accept_sock, const ch
     sock_protocol = BIO_ADDRINFO_protocol(res);
     sock_address = BIO_ADDRINFO_address(res);
     next = BIO_ADDRINFO_next(res);
+#ifdef AF_INET6
     if (sock_family == AF_INET6)
         sock_options |= BIO_SOCK_V6_ONLY;
     if (next != NULL
@@ -251,6 +254,7 @@ int do_server(int *accept_sock, const ch
             sock_options &= ~BIO_SOCK_V6_ONLY;
         }
     }
+#endif
 
     asock = BIO_socket(sock_family, sock_type, sock_protocol, 0);
     if (asock == INVALID_SOCKET
