diff -up gcc-3.0.4/gcc/config/sparc/sol2.h.v7 gcc-3.0.4/gcc/config/sparc/sol2.h
--- gcc-3.0.4/gcc/config/sparc/sol2.h.v7	2012-09-29 14:24:02.560188500 +0200
+++ gcc-3.0.4/gcc/config/sparc/sol2.h	2012-09-29 16:37:52.318389600 +0200
@@ -75,8 +75,13 @@ Boston, MA 02111-1307, USA.  */
 #undef ASM_CPU_DEFAULT_SPEC
 #define ASM_CPU_DEFAULT_SPEC "-xarch=v8"
 #endif
+#if TARGET_CPU_DEFAULT == TARGET_CPU_v7
+#undef ASM_CPU_DEFAULT_SPEC
+#define ASM_CPU_DEFAULT_SPEC "-xarch=v7"
+#endif
 #undef ASM_CPU_SPEC
 #define ASM_CPU_SPEC "\
+%{mcpu=v7:-xarch=v7} \
 %{mcpu=v8:-xarch=v8} \
 %{mcpu=v8plus:-xarch=v8plus} \
 %{mcpu=ultrasparc:-xarch=v8plusa} \
