diff -up gcc-2.95.3/gcc/config/sparc/sol2.h.sparcv8 gcc-2.95.3/gcc/config/sparc/sol2.h
--- gcc-2.95.3/gcc/config/sparc/sol2.h.sparcv8	1998-12-16 22:13:27.000000000 +0100
+++ gcc-2.95.3/gcc/config/sparc/sol2.h	2012-09-29 14:20:43.101385300 +0200
@@ -54,8 +54,13 @@ Boston, MA 02111-1307, USA.  */
 #undef ASM_CPU_DEFAULT_SPEC
 #define ASM_CPU_DEFAULT_SPEC "-xarch=v8plusa"
 #endif
+#if TARGET_CPU_DEFAULT == TARGET_CPU_v8
+#undef ASM_CPU_DEFAULT_SPEC
+#define ASM_CPU_DEFAULT_SPEC "-xarch=v8"
+#endif
 #undef ASM_CPU_SPEC
 #define ASM_CPU_SPEC "\
+%{mcpu=v8:-xarch=v8} \
 %{mcpu=v8plus:-xarch=v8plus} \
 %{mcpu=ultrasparc:-xarch=v8plusa} \
 %{!mcpu*:%(asm_cpu_default)} \
diff -up gcc-2.95.3/gcc/config/sparc/sparc.h.sparcv8 gcc-2.95.3/gcc/config/sparc/sparc.h
--- gcc-2.95.3/gcc/config/sparc/sparc.h.sparcv8	1999-06-21 19:48:49.000000000 +0200
+++ gcc-2.95.3/gcc/config/sparc/sparc.h	2012-09-29 14:20:18.031140000 +0200
@@ -146,12 +146,16 @@ extern enum cmodel sparc_cmodel;
 #define CPP_CPU64_DEFAULT_SPEC ""
 #define ASM_CPU64_DEFAULT_SPEC ""
 
-#if TARGET_CPU_DEFAULT == TARGET_CPU_sparc \
- || TARGET_CPU_DEFAULT == TARGET_CPU_v8
+#if TARGET_CPU_DEFAULT == TARGET_CPU_sparc
 #define CPP_CPU32_DEFAULT_SPEC ""
 #define ASM_CPU32_DEFAULT_SPEC ""
 #endif
 
+#if TARGET_CPU_DEFAULT == TARGET_CPU_v8
+#define CPP_CPU32_DEFAULT_SPEC "-D__sparc_v8__"
+#define ASM_CPU32_DEFAULT_SPEC "-Av8"
+#endif
+
 #if TARGET_CPU_DEFAULT == TARGET_CPU_sparclet
 #define CPP_CPU32_DEFAULT_SPEC "-D__sparclet__"
 #define ASM_CPU32_DEFAULT_SPEC "-Asparclet"
