--- openssl-0.9.7c/Makefile.org.origc	2004-01-06 15:20:40.890005000 +0100
+++ openssl-0.9.7c/Makefile.org	2004-01-06 15:22:02.059996000 +0100
@@ -390,6 +390,7 @@
 		  MINUSZ='-z '; \
 		  (${CC} -v 2>&1 | grep gcc) > /dev/null && MINUSZ='-Wl,-z,'; \
 		  set -x; ${CC} ${SHARED_LDFLAGS} -G -dy -z text \
+		  	-Wl,-R,@LIBDIR@ \
 			-o lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
 			-h lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
 			$${MINUSZ}allextract lib$$i.a $${MINUSZ}defaultextract \
