diff -ur -x CVS openssl7/ssl/s3_srvr.c ossl7/ssl/s3_srvr.c
--- openssl7/ssl/s3_srvr.c	2004-01-14 23:02:49.000000000 +0000
+++ ossl7/ssl/s3_srvr.c	2004-02-19 13:22:16.000000000 +0000
@@ -1588,11 +1591,27 @@
 
 		n2s(p,i);
 		enc_ticket.length = i;
+
+		if (n < enc_ticket.length + 6)
+			{
+			SSLerr(SSL_F_SSL3_GET_CLIENT_KEY_EXCHANGE,
+				SSL_R_DATA_LENGTH_TOO_LONG);
+			goto err;
+			}
+
 		enc_ticket.data = (char *)p;
 		p+=enc_ticket.length;
 
 		n2s(p,i);
 		authenticator.length = i;
+
+		if (n < enc_ticket.length + authenticator.length + 6)
+			{
+			SSLerr(SSL_F_SSL3_GET_CLIENT_KEY_EXCHANGE,
+				SSL_R_DATA_LENGTH_TOO_LONG);
+			goto err;
+			}
+
 		authenticator.data = (char *)p;
 		p+=authenticator.length;
 

