--- openssl-0.9.7c/Makefile.org.orig	2003-10-16 12:29:55.227462000 +0200
+++ openssl-0.9.7c/Makefile.org	2003-10-16 12:33:36.449814000 +0200
@@ -10,6 +10,7 @@
 SHLIB_MAJOR=
 SHLIB_MINOR=
 SHLIB_EXT=
+SHLIB_SOVER=
 PLATFORM=dist
 OPTIONS=
 CONFIGURE_ARGS=
@@ -390,8 +391,9 @@
 		  MINUSZ='-z '; \
 		  (${CC} -v 2>&1 | grep gcc) > /dev/null && MINUSZ='-Wl,-z,'; \
 		  set -x; ${CC} ${SHARED_LDFLAGS} -G -dy -z text \
+		  	-R${INSTALLTOP}/lib \
 			-o lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
-			-h lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
+			-h lib$$i.so.${SHLIB_SOVER}} \
 			$${MINUSZ}allextract lib$$i.a $${MINUSZ}defaultextract \
 			$$libs ${EX_LIBS} -lc ) || exit 1; \
 		libs="-l$$i $$libs"; \
