diff -up wget-1.14/src/connect.c.ipv6 wget-1.14/src/connect.c
--- wget-1.14/src/connect.c.ipv6	2012-06-02 19:36:09.000000000 +0200
+++ wget-1.14/src/connect.c	2013-04-28 11:39:39.610656000 +0200
@@ -294,8 +294,10 @@ connect_to_ip (const ip_address *ip, int
        {
            if (ip->family == AF_INET)
                logprintf (LOG_VERBOSE, _("Connecting to %s:%d... "), txt_addr, port);
+#ifdef ENABLE_IPV6
            else if (ip->family == AF_INET6)
                logprintf (LOG_VERBOSE, _("Connecting to [%s]:%d... "), txt_addr, port);
+#endif
        }
     }
 
diff -up wget-1.14/src/http.c.ipv6 wget-1.14/src/http.c
--- wget-1.14/src/http.c.ipv6	2012-07-07 10:57:20.000000000 +0200
+++ wget-1.14/src/http.c	2013-04-28 11:56:49.332090000 +0200
@@ -1962,11 +1962,13 @@ gethttp (struct url *u, struct http_stat
           int family = socket_family (pconn.socket, ENDPOINT_PEER);
           sock = pconn.socket;
           using_ssl = pconn.ssl;
+#ifdef ENABLE_IPV6
           if (family == AF_INET6)
              logprintf (LOG_VERBOSE, _("Reusing existing connection to [%s]:%d.\n"),
                         quotearg_style (escape_quoting_style, pconn.host),
                          pconn.port);
           else
+#endif
              logprintf (LOG_VERBOSE, _("Reusing existing connection to %s:%d.\n"),
                         quotearg_style (escape_quoting_style, pconn.host),
                         pconn.port);
