diff -up coreutils-8.20/Makefile.in.orig coreutils-8.20/Makefile.in
--- coreutils-8.20/Makefile.in.orig	2013-02-01 21:59:38.269116000 +0100
+++ coreutils-8.20/Makefile.in	2013-02-02 19:37:19.327331100 +0100
@@ -3114,7 +3114,7 @@ src_expand_LDADD = $(LDADD)
 
 # for various GMP functions
 src_expr_LDADD = $(LDADD) $(LIB_GMP)
-src_factor_LDADD = $(LDADD) $(LIB_GMP)
+src_factor_LDADD = $(LDADD) $(LIB_GMP) $(LIBICONV)
 src_false_LDADD = $(LDADD)
 src_fmt_LDADD = $(LDADD)
 src_fold_LDADD = $(LDADD)
